device/tests/Microsoft.Azure.Devices.Client.Test/IoTHubClientDiagnosticTest.cs

```diff
@@ -0,0 +1,188 @@
+﻿namespace Microsoft.Azure.Devices.Client.Test
+{
+    using System;
+    using System.Collections.Specialized;
+    using System.Globalization;
+    using System.IO;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Web;
+    using Microsoft.Azure.Amqp;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class IoTHubClientDiagnosticTest
+    {
+        [TestMethod]
+        [TestCategory("IoTHubClientDiagnostic")]
+        public void IoTHubClientDiagnostic_AddDiagnosticInfoIfNecessary_Test()
+        {
+            Message message = this.CreateMessage();
+            const int DiagPercentageWithoutDiagnostic = 0;
+            const int DiagPercentageWithDiagnostic = 100;
+
+            int messageCount = 0;
+            IoTHubClientDiagnostic.AddDiagnosticInfoIfNecessary(message, DiagPercentageWithoutDiagnostic, ref messageCount);
+
+            Assert.IsFalse(message.SystemProperties.ContainsKey(MessageSystemPropertyNames.DiagId));
+            Assert.IsFalse(message.SystemProperties.ContainsKey(MessageSystemPropertyNames.DiagCorrelationContext));
+
+            message = this.CreateMessage();
```

Comment:
Nit: Avoid using `this.` (we're trying to formalize coding standards similar to the ones in dotnet/corefx).

Reply:
fixed